src/solver/conv_hip_implicit_gemm_wrw_v4r4_xdlops_padded_gemm.cpp

```diff
@@ -0,0 +1,1106 @@
+          PerformanceImplicitGemmWrwV4R4Xdlops_Padded_Gemm(
+              64, 64, 2, 32, 32, 4, 16, 64, 16, false, false)
+{
+}
+
+PerformanceImplicitGemmWrwV4R4Xdlops_Padded_Gemm::PerformanceImplicitGemmWrwV4R4Xdlops_Padded_Gemm(
+    int GemmMPerBlock_,
+    int GemmNPerBlock_,
+    int GemmKPerBlock_,
+    int GemmMPerWave_,
+    int GemmNPerWave_,
+    int GemmKPack_,
+    int GemmMFactor_,
+    int GemmNFactor_,
+    int GemmKFactor_,
+    bool GemmAThreadCopyMoreGemmK_,
+    bool GemmBThreadCopyMoreGemmKPack_)
+    : GemmMPerBlock(GemmMPerBlock_),
+      GemmNPerBlock(GemmNPerBlock_),
+      GemmKPerBlock(GemmKPerBlock_),
+      GemmMPerWave(GemmMPerWave_),
+      GemmNPerWave(GemmNPerWave_),
+      GemmKPack(GemmKPack_),
+      GemmMFactor(GemmMFactor_),
+      GemmNFactor(GemmNFactor_),
+      GemmKFactor(GemmKFactor_),
+      GemmAThreadCopyMoreGemmK(GemmAThreadCopyMoreGemmK_),
+      GemmBThreadCopyMoreGemmKPack(GemmBThreadCopyMoreGemmKPack_)
+{
+}
+
+bool PerformanceImplicitGemmWrwV4R4Xdlops_Padded_Gemm::
+operator==(const PerformanceImplicitGemmWrwV4R4Xdlops_Padded_Gemm& other) const
+{
+    // clang-format off
+    return GemmMPerBlock == other.GemmMPerBlock
+        && GemmNPerBlock == other.GemmNPerBlock
+        && GemmKPerBlock == other.GemmKPerBlock
+        && GemmMPerWave == other.GemmMPerWave
+        && GemmNPerWave == other.GemmNPerWave
+        && GemmKPack == other.GemmKPack
+        && GemmMFactor == other.GemmMFactor
+        && GemmNFactor == other.GemmNFactor
+        && GemmKFactor == other.GemmKFactor
+        && GemmAThreadCopyMoreGemmK  == other.GemmAThreadCopyMoreGemmK
+        && GemmBThreadCopyMoreGemmKPack  == other.GemmBThreadCopyMoreGemmKPack;
+    // clang-format on
+}
+
+bool PerformanceImplicitGemmWrwV4R4Xdlops_Padded_Gemm::SetNextValue()
+{
+    do
+    {
+        // list performance parameters in reverse order, in order for tuning to iterate over the
+        // range in normal order
+        if(!NextFlag<false, true>(GemmBThreadCopyMoreGemmKPack))
+            break;
+        if(!NextFlag<false, false>(GemmAThreadCopyMoreGemmK))
+            break;
+        if(!NextTwoPower<1, 8>(GemmKPack))
+            break;
+        if(!NextTwoPower<32, 128>(GemmNPerWave))
+            break;
+        if(!NextTwoPower<32, 128>(GemmMPerWave))
+            break;
+        if(!NextTwoPower<2, 8>(GemmKPerBlock))
+            break;
+        if(!NextTwoPower<64, 256>(GemmNPerBlock))
+            break;
+        if(!NextTwoPower<64, 256>(GemmMPerBlock))
+            break;
+        return false;
+    } while(false);
+
+    return true;
+}
+
+void PerformanceImplicitGemmWrwV4R4Xdlops_Padded_Gemm::EuristicInit(const ConvolutionContext& ctx)
+{
+    PerformanceImplicitGemmWrwV4R4Xdlops_Padded_Gemm tmp;
+
+    // loop over certain ranges of tuning parameter
+    auto get_euristic_config = [&](auto is_valid_func) {
+        if(ctx.IsFp32())
+        {
+            tmp = {256, 256, 8, 128, 128, 4, 16, 64, 16, false, true};
```

Comment:
Please add comment here that GemmMFactor GemmNFactor, GemmKFactor are fixed value at this moment.